src/config/passport.js

```diff
@@ -0,0 +1,25 @@
+const passport = require('passport');
+const GithubStrategy = require('passport-github').Strategy;
+const User = require('../models/User');
+const config = require('./index');
+
+passport.use(new GithubStrategy(config.githubOAuth, 
+  (accessToken, refreshToken, profile, cb) => {
+    User
+      .findOneOrCreate(accessToken, profile)
+      .then((user) => {
+        if(user) 
+          cb(null, user);
+        else
+          cb();
+      })
+}))
+
+passport.serializeUser((user, done) => done(null, user.id));
+
+passport.deserializeUser((userID, done) => {
```

Comment:
Check if this is possible to use async await instead of callback in whole file

Reply:
No, passport allow only callback method.